### PR TITLE
Offline loading fix

### PR DESCRIPTION
**Pull request by tunjid**
Addresses #169

### CODE REVIEW

**Comment by dlam** [comment]
It's probably worth asking... in what cases do we actually want to prevent the retry button from showing up?

Technically not having internet access is an error that you might want to retry.. what happens if you search a new query, but you lose internet for a bit due to a poor connectivity.. should that trigger the retry button to show or should it hide because we still have the previous results? I think it's confusing if you don't get an error prompt in the UI, but your search actually failed (I also think the same can be said of the offline in initial load case).

I think overall we might be better off not including the retry button change, but always showing the list makes sense to me, as long as it looks fine in UI with the retry button overlaid ontop of it.

**Comment by tunjid** [comment]
@dlam what do you think of moving the retry button the the `Toolbar`? It looks weird when the list is visible and the retry button is overlaid on it.

**Comment by florina-muntenescu** [comment]
> @dlam what do you think of moving the retry button the the `Toolbar`? It looks weird when the list is visible and the retry button is overlaid on it.

The list shouldn't be visible and the retry button shown on top, that's for sure.
But at the same time, the retry button should be close to the UI that needs it, so putting it in the Toolbar feels too far.

> With the above the retry button only shows if the search fails and the list is empty, so if you try a new search that has no cached result, it will still show up.

**Comment by tunjid** [comment]
@florina-muntenescu @dlam would a dialog be too intrusive? Another benefit the dialog can have is we can communicate the error better; it can say for example "Failed fetching from remote, here's your cached version instead" or something along those lines.

**Comment by dlam** [comment]
I'm not a sure a dialog is great because it can be dismissed and then you lose the ability to retry as a user.

What about just a footer / header? Paging has a built-in for this too in LoadStateAdapter.

**Comment by tunjid** [comment]
@dlam @florina-muntenescu what do y'all think? Retry button if the list is empty, otherwise a header with the same functionality.

**Comment by dlam** [comment]
That seems fine to me as long as we are okay with the added complexity to the codelab. I wish we could also just do a header in all cases to make it simpler, but I guess that conflicts with hiding the list :|

**Review by dlam — APPROVED** [review]

File: app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt [hunk 1]
```diff
@@ -81,22 +82,30 @@ class SearchRepositoriesActivity : AppCompatActivity() {
...
 
     private fun initAdapter() {
-        binding.list.adapter = adapter.withLoadStateHeaderAndFooter(
-            header = ReposLoadStateAdapter { adapter.retry() },
-            footer = ReposLoadStateAdapter { adapter.retry() }
-        )
+        val header = ReposLoadStateAdapter { adapter.retry() }
+        val footer = ReposLoadStateAdapter { adapter.retry() }
+
+        binding.list.adapter = ConcatAdapter(header, adapter, footer)
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt:88]: why do we need this and we can't use `withLoadStateHeaderAndFooter`? The method should do something similar to what you're doing here.

**Pull request merged**